packages/language-server/test/completion-items-spec.ts

```diff
@@ -251,6 +252,189 @@ describe("the UI5 language assistant Code Completion Services", () => {
     expect(suggestionKinds).to.deep.equal([CompletionItemKind.Text]);
   });
 
+  it("will replace the aggregation closing tag name when the tag is closed and has the same name as the opening tag", () => {
+    const xmlSnippet = `<mvc:View 
+        xmlns:mvc="sap.ui.core.mvc"
+        xmlns="sap.m">
+      <List>
+        <te⇶></⭲te⭰>
+      </List>
+    </mvc:View>`;
+    const suggestions = getSuggestions(xmlSnippet, ui5SemanticModel);
+    const suggestionsDetails = map(suggestions, suggestion => ({
+      label: suggestion.label,
+      tagName: getTagName(suggestion.textEdit),
+      additionalTextEdits: suggestion.additionalTextEdits,
+      replacedText: getTextInRange(xmlSnippet, suggestion.textEdit?.range)
+    }));
+    const suggestionKinds = uniq(
+      map(suggestions, suggestion => suggestion.kind)
+    );
+
+    const ranges = getRanges(xmlSnippet);
+    expect(ranges, "additional text edits ranges").to.have.lengthOf(1);
+
+    expect(suggestionsDetails).to.deep.equalInAnyOrder([
+      {
+        label: "contextMenu",
+        tagName: "contextMenu",
+        additionalTextEdits: [
+          {
+            range: ranges[0],
+            newText: `contextMenu`
+          }
+        ],
+        replacedText: "te"
+      },
+      {
+        label: "items",
+        tagName: "items",
+        additionalTextEdits: [
+          {
+            range: ranges[0],
+            newText: `items`
+          }
+        ],
+        replacedText: "te"
+      },
+      {
+        label: "swipeContent",
+        tagName: "swipeContent",
+        additionalTextEdits: [
+          {
+            range: ranges[0],
+            newText: `swipeContent`
+          }
+        ],
+        replacedText: "te"
+      }
+    ]);
+
+    expect(suggestionKinds).to.deep.equal([CompletionItemKind.Text]);
+  });
+
+  it("will replace the class closing tag name when the tag is closed and has a different name from the opening tag", () => {
```

Comment:
maybe we should not replace in this case.
Basically be very careful to only replace when both tags are aligned to reduce
the chance we are replacing on an unrelated tag.

This is similar to how I think tag syncing in XML-Tools should happen.
I prefer to be 99.99% correct when replacing at the cost of only editing for 90% of use cases.

Reply:
Ok, changed